### PR TITLE
Fix #1951: hints_summary.xml optimised

### LINKED ISSUE #1951: Optimise hints_summary.xml
This issue is tracking the optimisation work of `hints_summary.xml`

1. Make sure that the layout uses `ConstraintLayout` and there is minimal nesting of views. If this view already uses `ConstraintLayout` then it's fine.

2. Make sure that all textviews use the style from `styles.xml` file like `Heading1`, `TextFieldLabel`, `Body`, etc. whichever suits best.

While making the above changes the UI should not get affected and the end result should still be same as before (or an optimised version of UI).

If there are multiple files with the same name, update them all.

## References
Mobile Portrait: https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/grid/
Mobile Landscape: https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/grid/
Tablet: https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/grid/
Wiki: https://github.com/oppia/oppia-android/wiki/Working-on-UI

### PR DESCRIPTION
**Pull request by arinarora18**
## Explanation

Fixes #1951: This PR optimises the hints_summary.xml for mobile and tablet devices for both portrait and landscape orientation. 
 
## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by arinarora18** [comment]
Portrait before and after:
Portrait before
Portrait after

Landscape before and after:
Landscape before
Landscape after

Portrait before and after(tablet):
Portrait before (tablet)
Portrait after(tablet)

Landscape before and after(tablet):
Landscape before(tablet)
Landscape after(tablet)

**Comment by arinarora18** [comment]
@anandwana001 any updates?

File: .idea/misc.xml [hunk 1]
```diff
@@ -1,9 +1,20 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="DesignSurface">
```
Comment (anandwana001) [inline .idea/misc.xml:3]: please don't make any changes to this file.

File: .idea/runConfigurations.xml [hunk 2]
```diff
@@ -1,12 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<project version="4">
-  <component name="RunConfigurationProducerService">
-    <option name="ignoredProducers">
-      <set>
-        <option value="org.jetbrains.plugins.gradle.execution.test.runner.AllInPackageGradleConfigurationProducer" />
-        <option value="org.jetbrains.plugins.gradle.execution.test.runner.TestClassGradleConfigurationProducer" />
-        <option value="org.jetbrains.plugins.gradle.execution.test.runner.TestMethodGradleConfigurationProducer" />
-      </set>
```
Comment (anandwana001) [inline .idea/runConfigurations.xml:9]: please don't make any changes to this file.

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Thanks @arinarora18 for your first PR. 

Always add screenshots in the PR description.

Rather than updating all the UI files in terms of views, could you follow an approach where we first can look at all the common things like margins, padding, and then you can see the view component tree in the android studio on in layout editor if views are same and their margin, padding is different then fetch the margin and padding from the respective dimension file. 

https://developer.android.com/guide/topics/resources/providing-resources#AlternativeResources

**Review by rt4914 — CHANGES_REQUESTED** [review]
@arinarora18 I suggested not doing this PR directly as it will make this really hard to review.

Instead I suggested first focus on this issue https://github.com/oppia/oppia-android/issues/3537 and solve this first. Once it gets solved, then solve this issues.

Reason behind this suggestions:
Currently develop contains 4 files for `hints_summary.xml`.
Once you solve #3537 and it gets merged you will have only 1 file in develop.
Now solve that one single file and optimising it, is much more easier.

Also please have look at [this](https://github.com/oppia/oppia-android/wiki/Working-on-UI) too. 

Thanks.

**Pull request closed without merging**